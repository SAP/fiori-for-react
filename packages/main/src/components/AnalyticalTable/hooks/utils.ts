export const makeTemplateColumns = (columns, resizedColumns) =>
  columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
