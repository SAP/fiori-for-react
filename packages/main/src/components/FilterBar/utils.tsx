import { cloneElement, ReactElement } from 'react';

export const setPropsOfChildren = (children, currentRef) => {
  return children.map((child) => {
    if (child.props.children.props[currentRef]?.children) {
      const childrenOfChild = child.props.children;
      if (
        child.props.children.props[currentRef]?.tagName === 'UI5-SELECT' ||
        child.props.children.props[currentRef]?.tagName === 'UI5-MULTI-COMBOBOX'
      ) {
        const selectedIndices = Array.from(child.props.children.props[currentRef].children)
          .map((item, index) => (item._state.selected ? index : false))
          .filter((el) => el !== false);
        const options = child.props.children.props.children.map((item, index) => {
          if (selectedIndices.includes(index)) {
            return cloneElement(item, { selected: true });
          }
          return cloneElement(item, { selected: false });
        });
        return cloneElement(child, {
          children: { ...childrenOfChild, props: { ...childrenOfChild.props, children: options } }
        });
      }
      if (
        child.props.children.props[currentRef]?.tagName === 'UI5-INPUT' ||
        child.props.children.props[currentRef]?.tagName === 'UI5-DATEPICKER'
      ) {
        return cloneElement(child, {
          children: {
            ...childrenOfChild,
            props: { ...childrenOfChild.props, value: child.props.children.props[currentRef]._state.value }
          }
        });
      }
      if (
        child.props.children.props[currentRef]?.tagName === 'UI5-SWITCH' ||
        child.props.children.props[currentRef]?.tagName === 'UI5-CHECKBOX'
      ) {
        return cloneElement(child, {
          children: {
            ...childrenOfChild,
            props: { ...childrenOfChild.props, checked: child.props.children.props[currentRef]._state.checked }
          }
        });
      }
      return child;
    }
    return child;
  });
};

export const addRef = (children, refsArray, currentRef) =>
  children.map((child) => {
    const childRef = refsArray.filter((item) => item.key === child.key);
    const childrenAttributes = child.props.children;
    const childrenProps = child.props.children.props;
    return cloneElement(child as ReactElement<any>, {
      children: { ...childrenAttributes, props: { ...childrenProps, [currentRef]: childRef[0]?.node } }
    });
  });

export const renderSearchWithValue = (searchElement, searchValue) => {
  return cloneElement(searchElement(), { value: searchValue });
};
