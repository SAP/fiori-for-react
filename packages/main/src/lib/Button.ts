import { Button, ButtonPropTypes } from '../webComponents/Button';

export { Button, ButtonPropTypes };
