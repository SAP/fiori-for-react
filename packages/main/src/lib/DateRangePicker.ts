import { DateRangePicker } from '../webComponents/DateRangePicker';

export { DateRangePicker };
