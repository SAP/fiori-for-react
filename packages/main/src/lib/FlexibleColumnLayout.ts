import { FlexibleColumnLayout } from '../webComponents/FlexibleColumnLayout';

export { FlexibleColumnLayout };
