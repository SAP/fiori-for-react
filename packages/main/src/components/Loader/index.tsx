import { StyleClassHelper } from '@ui5/webcomponents-react-base/lib/StyleClassHelper';
import React, { forwardRef, RefObject, FC, CSSProperties } from 'react';
import { createUseStyles } from 'react-jss';
import { CommonProps } from '../../interfaces/CommonProps';
import { LoaderType } from '@ui5/webcomponents-react/lib/LoaderType';
import { styles } from './Loader.jss';

export interface LoaderProps extends CommonProps {
  type?: LoaderType;
  progressBarWidth?: CSSProperties['width'];
}

const useStyles = createUseStyles<JSSTheme, keyof ReturnType<typeof styles>>(styles, { name: 'Loader' });

const Loader: FC<LoaderProps> = forwardRef((props: LoaderProps, ref: RefObject<HTMLDivElement>) => {
  const { className, type, progressBarWidth, tooltip, slot, style } = props;
  const classes = useStyles(props);

  const loaderClasses = StyleClassHelper.of(classes.loader);
  if (className) {
    loaderClasses.put(className);
  }
  loaderClasses.put(classes[`loader${type}`]);

  return (
    <div
      ref={ref}
      className={loaderClasses.valueOf()}
      data-component-name="Loader"
      aria-busy="true"
      role="progressbar"
      title={tooltip || 'Please wait'}
      slot={slot}
      style={style}
    />
  );
});

Loader.defaultProps = {
  type: LoaderType.Indeterminate,
  progressBarWidth: '0px'
};

Loader.displayName = 'Loader';

export { Loader };
