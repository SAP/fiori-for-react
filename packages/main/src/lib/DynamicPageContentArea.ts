import { DynamicPageContentArea } from '../components/DynamicPageContentArea';

export { DynamicPageContentArea };
