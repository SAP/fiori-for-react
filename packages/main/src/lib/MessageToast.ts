import { MessageToast } from '../components/MessageToast';

export { MessageToast };

// Required for React lazy
export default MessageToast;
