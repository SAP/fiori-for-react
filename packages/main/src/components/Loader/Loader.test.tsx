import { renderThemedComponent } from '@shared/tests/utils';
import React from 'react';
import { LoaderType } from '@ui5/webcomponents-react/lib/LoaderType';
import { Loader } from '@ui5/webcomponents-react/lib/Loader';

describe('Loader', () => {
  test('small', () => {
    const wrapper = renderThemedComponent(<Loader type={LoaderType.Indeterminate} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('medium', () => {
    const wrapper = renderThemedComponent(<Loader type={LoaderType.Determinate} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('large', () => {
    const wrapper = renderThemedComponent(<Loader type={LoaderType.Determinate} progressBarWidth="42%" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('with custom class name', () => {
    const wrapper = renderThemedComponent(<Loader />);
    expect(wrapper).toMatchSnapshot();
  });
});
