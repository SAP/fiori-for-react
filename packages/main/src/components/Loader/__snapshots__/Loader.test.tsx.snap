// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader large 1`] = `
<div
  aria-busy="true"
  class="Loader--loader- Loader--loader-0- Loader--loaderDeterminate-"
  data-component-name="Loader"
  role="progressbar"
  title="Please wait"
/>
`;

exports[`Loader medium 1`] = `
<div
  aria-busy="true"
  class="Loader--loader- Loader--loader-0- Loader--loaderDeterminate-"
  data-component-name="Loader"
  role="progressbar"
  title="Please wait"
/>
`;

exports[`Loader small 1`] = `
<div
  aria-busy="true"
  class="Loader--loader- Loader--loader-0- Loader--loaderIndeterminate-"
  data-component-name="Loader"
  role="progressbar"
  title="Please wait"
/>
`;

exports[`Loader with custom class name 1`] = `
<div
  aria-busy="true"
  class="Loader--loader- Loader--loader-0- Loader--loaderIndeterminate-"
  data-component-name="Loader"
  role="progressbar"
  title="Please wait"
/>
`;
